streaming/src/main/scala/org/apache/spark/streaming/dstream/SocketInputDStream.scala

```diff
@@ -52,7 +52,13 @@ class SocketReceiver[T: ClassTag](
     storageLevel: StorageLevel
   ) extends Receiver[T](storageLevel) with Logging {
 
+  private var socket: Socket = null
+
   def onStart() {
+    logInfo(s"Connecting to $host:$port")
+    socket = new Socket(host, port)
+    logInfo(s"Connecting to $host:$port")
```

Comment:
Log message is wrong -- it's connected now. You could omit this message anyway. Don't you need to move handling of `ConnectException` here? and finally, nit, but I think initializing the socket value to "_" is more canonical in Scala? if you initialize at all.